src/Sarif/Writers/SarifOneZeroZeroLogger.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Microsoft.CodeAnalysis.Sarif.VersionOne;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif.Writers
+{
+    /// <summary>
+    /// A SARIF logger that works by generating a SARIF v2 log file which is transformed into 
+    /// SARIF v1 when the instance is disposed. The file location used to produced the preliminary
+    /// v2 content is overwritten in place to produce the transformed v1 file.
```

Comment:
>file [](start = 73, length = 4)

That is awfully clever :-) And clever code needs a good comment, and this is one.